### PR TITLE
fix(Breadcrumb Item): support Breadcrumb Item onClick and className

### LINKED ISSUE #41274: 给面包屑添加onClick事件后，点击不触发事件
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-k8uc82?file=/index.js)

### Steps to reproduce

点击 Test01、Test02或Test03 都不触发onClick事件

### What is expected?

触发onClick事件，控制台输出结果

### What is actually happening?

不触发onClick事件，控制到无输出

| Environment | Info |
| --- | --- |
| antd | 5.3.1 |
| React | 18.1.0 |
| System | windows10 |
| Browser | Google Chrome	111.0.5563.65 (Official Build) (64-bit) (cohort: Stable) |

### PR DESCRIPTION
**Pull request by notverysalty**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #41274
close #41283

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     support Breadcrumb Item onClick and className      |
| 🇨🇳 Chinese |     Breadcrumb Item支持onClick和className     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
Duplicate of #41283 ？

**Comment by notverysalty** [comment]
> rebase please

ok, thanks😂

File: components/breadcrumb/__tests__/Breadcrumb.test.tsx [hunk 1]
```diff
@@ -329,4 +329,32 @@ describe('Breadcrumb', () => {
     expect(errSpy).not.toHaveBeenCalled();
     errSpy.mockRestore();
   });
+
+  // https://github.com/ant-design/ant-design/issues/41274
+  it('should support Breadcrumb Item onClick and className', () => {
+    const onClick = jest.fn();
```
Comment (yoyo837) [inline components/breadcrumb/__tests__/Breadcrumb.test.tsx:335]: 这个用例需要完善

Comment (yoyo837) [reply components/breadcrumb/__tests__/Breadcrumb.test.tsx:335]: 需要检测点击有触发, 样式要检查存在

Comment (notverysalty) [reply components/breadcrumb/__tests__/Breadcrumb.test.tsx:335]: 好的，谢谢大佬😭

File: components/breadcrumb/__tests__/Breadcrumb.test.tsx [hunk 2]
```diff
@@ -329,4 +329,32 @@ describe('Breadcrumb', () => {
...
+            onClick,
+            className: 'chinese',
+          },
+          {
+            title: 'An Application',
+          },
+        ]}
+      />,
+    );
+    expect(container.firstChild).toMatchSnapshot();
```
Comment (heiyu4585) [inline components/breadcrumb/__tests__/Breadcrumb.test.tsx:358]: 请补充 类似代码
```
    fireEvent.click(XXXX);
    expect(onClick).toHaveBeenCalledWith();
    expect(XXXXXX).toHaveClass('chinese');
```

Comment (notverysalty) [reply components/breadcrumb/__tests__/Breadcrumb.test.tsx:358]: 好的，谢谢黑雨大佬😭

**Comment by zombieJ** [comment]
撞车了 #41283

**Comment by zombieJ** [comment]
> 撞车了 #41283

上面这个把 onClick 和 className 都补上了，这个 PR 就先关了哈~

**Comment by notverysalty** [comment]
> 

好滴😋

**Pull request closed without merging**